modules/checklist/steps-manager.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Elementor\Modules\Checklist;
+
+use Elementor\Modules\Checklist\Steps\Create_Pages;
+use Elementor\Modules\Checklist\Steps\Step_Base;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Steps_Manager {
+	/** @var Step_Base[] $step_instances */
+	private $step_instances = [];
+
+	private Module $module;
+
+	public function __construct( Module $module ) {
+		$this->module = $module;
+		$this->register_steps();
+	}
+
+	/**
+	 * Gets formatted and ordered array of step ( step data, is_marked_done and is_completed )
+	 *
+	 * @return array
+	 */
+	public function get_steps_for_frontend() : array {
+		$formatted_steps = [];
+
+		foreach ( $this->step_instances as $step ) {
+			$formatted_steps[] = $step->get_step_data();
+		}
+
+		return $formatted_steps;
+	}
+
+	/**
+	 * Marks a step as done, returns true if the step was found and marked or false otherwise
+	 *
+	 * @param string $step_id
+	 *
+	 * @return bool
+	 */
+	public function mark_step_as_done( $step_id ) : bool {
+		foreach ( $this->step_instances as $step ) {
+			if ( $step->get_id() === $step_id ) {
+				$step->mark_as_done();
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Unmarks a step as done, returns true if the step was found and unmarked or false otherwise
+	 *
+	 * @param string $step_id
+	 *
+	 * @return bool
+	 */
+	public function unmark_step_as_done( $step_id ) : bool {
+		foreach ( $this->step_instances as $step ) {
+			if ( $step->get_id() === $step_id ) {
+				$step->unmark_as_done();
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Maybe marks a step as completed (depending on if source allows it), returns true if the step was found and marked or false otherwise
+	 *
+	 * @param $step_id
+	 *
+	 * @return bool
+	 */
+	public function maybe_mark_step_as_complete( $step_id ) : bool {
```

Comment:
a bit confusing between `done` and `complete` namings

Reply:
I see, I started a discussion on slack and we'll decide on better naming